### DISCUSSION TITLE
Geolocation not working with rootless Podman

### DISCUSSION THREAD

**Discussion by jurf** [plausible/analytics#1410] [Self-Hosted Support] [answered]
**Edit**: Originally, I thought this was caused by removing the GeoLite2. However, around the time, I also moved to a rootless Podman setup, which turned out to be the cause of the problem, so I edited the title.

- - -

Hi,

after reading that GeoLite2 was no longer needed, I removed it from my setup to simplify it. I was careful to remove all related variables; the UI shows DB-IP is used, but no data is present. Is there anything I can do to fix my setup?

**Comment by ukutaht** [comment]
What docker tag are you using?

**Comment by ukutaht** [comment]
Similar issue has been reported with the `latest` tag. You might want to try the pre-release `v1.4.0-rc.0` tag which should fix the issue and also includes a bunch of new stuff. See https://github.com/plausible/analytics/discussions/1409

**Reply by jurf** [reply]
Yes, I am using the `latest` tag. I will try with the pre-release; thanks!

**Reply by jurf** [reply]
I have tried the release candidate and now the regular release, neither gave me country data.

**Reply by ukutaht** [reply]
I just tested the release candidate and it gets the country data for me. Does the file exist in your container?

```
docker exec -it <your-plausuble-container-name> ls lib/plausible-0.0.1/priv/geodb/dbip-country.mmdb
```

**Reply by jurf** [reply]
Yes.
```
$ du -h lib/plausible-0.0.1/priv/geodb/dbip-country.mmdb
3.2M	lib/plausible-0.0.1/priv/geodb/dbip-country.mmdb
$ sha256sum lib/plausible-0.0.1/priv/geodb/dbip-country.mmdb
8a1c748e551445712cefbbb59d2ba8d2ac2b01db6c1760c26e3ad982be7d3555  lib/plausible-0.0.1/priv/geodb/dbip-country.mmdb
```

**Reply by ukutaht** [reply]
OK. Next step: is your application config pointing to the right file? Should look like this.
```
➜  analytics git:(master) ✗ docker exec -it plausible /app/bin/plausible remote                                          

Erlang/OTP 24 [erts-12.0] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit:no-native-stack]

Interactive Elixir (1.12.2) - press Ctrl+C to exit (type h() ENTER for help)
iex(plausible@UkuLenovo)1> Application.get_env(:geolix, :databases)
[
  %{
    adapter: Geolix.Adapter.MMDB2,
    id: :country,
    result_as: :raw,
    source: "/app/lib/plausible-0.0.1/priv/geodb/dbip-country.mmdb"
  }
]
```

**Reply by jurf** [reply]
That doesn’t seem to work…
```
$ /app/bin/plausible remote
Erlang/OTP 24 [erts-12.0] [source] [64-bit] [smp:1:1] [ds:1:1:10] [async-threads:1] [jit:no-native-stack]

Could not contact remote node plausible@7c379a00fbe9, reason: :nodedown. Aborting...
```

**Reply by jurf** [reply]
I recreated the container; the command now launches correctly. The output seems fine.
```
[
  %{
    adapter: Geolix.Adapter.MMDB2,
    id: :country,
    result_as: :raw,
    source: "/app/lib/plausible-0.0.1/priv/geodb/dbip-country.mmdb"
  }
]
```

**Reply by jurf** [reply]
Hi, this problem persists. Is there anything else I can do?

**Reply by ukutaht** [reply]
Sorry, your config seems correct and I don't have any other reports of this failing. Not sure what else to try

**Reply by jurf** [reply]
Can I somehow reset the app without losing the already-gathered statistics?

**Reply by ukutaht** [reply]
There's no easy way AFAIK.
1. Back up the databases (Postgres, Clickhouse)
2. Nuke everything
3. Start from scratch
4. Restore databases

**Reply by jurf** [reply]
I just tried returning the Maxmind database but it did not fix the issue.

**Comment by jurf** [comment, marked as answer]
I found the cause.

The issue was caused by rootless podman, which currently does not forward the real source IP by default.

The workaround mentioned in the [docs](https://docs.podman.io/en/latest/markdown/podman-create.1.html#network-mode-net) does not work for caddy-gen. However, because docker-compose uses networks by default, which in turn the mentioned workaround does not support (https://github.com/containers/podman/issues/8193), the only way to do this currently is to expose the ports of the containers and use the `"host"` network mode.

To avoid changing the caddy-gen template, one can use this on the containers as a quick and dirty hack:

```
    ports:
      - "8000:8000"
    labels:
      virtual.proxy.directives: "to 127.0.0.1:8000"
      virtual.proxy.lb_policy: "first"
```

The load-balance policy is needed as this only adds another upstream, the caddy-gen-generated one is still there but will never be used.

I hope someone finds this information useful.
